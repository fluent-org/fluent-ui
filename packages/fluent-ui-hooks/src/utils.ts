export const global = typeof window !== 'undefined' && window
