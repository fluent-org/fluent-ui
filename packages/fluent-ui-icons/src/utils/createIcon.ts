import * as React from 'react'
import styled from 'styled-components'

export type Children = [string, any]
export type JSX = ['svg', any, Children]

function createIcon(jsx: JSX, componentName: string): React.ForwardRefExoticComponent<{}> {
  const [type, props, children] = jsx
  function Icon(_props: any): any {
    return React.createElement(type, { ...props, ..._props }, React.createElement(...children))
  }
  const SvgIconStyled = styled(Icon)`
    width: 0.83em;
    height: 0.83em;
    display: inline-block;
    user-select: none;
    font-size: inherit;
    color: inherit;
    fill: currentColor;
  `
  SvgIconStyled.displayName = componentName
  return SvgIconStyled
}

export default createIcon
