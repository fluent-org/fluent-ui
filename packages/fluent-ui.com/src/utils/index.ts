export * from './scroll'
export * from './theme'
