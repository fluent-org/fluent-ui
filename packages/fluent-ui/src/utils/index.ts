export { isObject, deepMerge } from './deepMerge'
export { default as omit } from './omit'
export * from './mixins'
