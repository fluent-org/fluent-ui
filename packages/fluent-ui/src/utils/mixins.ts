export const findAllByType = (component: any, type: string): any[] => {
  let matches: any[] = []

  if (component.type === type) {
    matches.push(component)
  }

  if (component.children) {
    component.children.forEach(
      (child: any): void => {
        matches = matches.concat(findAllByType(child, type))
      }
    )
  }

  return matches
}
