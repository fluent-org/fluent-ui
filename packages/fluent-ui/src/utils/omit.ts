interface Obj {
  [key: string]: any
}

function omit(obj: Obj, fields: string[]): Omit<Obj, string> {
  const shallowCopy: Obj = {
    ...obj
  }
  for (let i = 0; i < fields.length; i++) {
    const key: string = fields[i]
    delete shallowCopy[key]
  }
  return shallowCopy
}

export default omit
