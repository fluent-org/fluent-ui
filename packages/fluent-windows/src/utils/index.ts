export * from './mixins'
export * from './convertingColor'
