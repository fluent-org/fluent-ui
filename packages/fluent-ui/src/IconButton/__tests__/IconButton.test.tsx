import * as React from 'react'
import * as renderer from 'react-test-renderer'
import 'jest-styled-components'

import { Accept as AcceptIcon } from '@fluent-ui/icons'
import { ThemeProvider, IconButton } from '../..'
import { findAllByType } from '../../utils'

describe('IconButton', (): void => {
  const theme = {}
  test('basic', (): void => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <IconButton>
          <AcceptIcon />
        </IconButton>
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('variant', (): void => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <IconButton variant="primary">
          <AcceptIcon />
        </IconButton>
        <IconButton variant="accent">
          <AcceptIcon />
        </IconButton>
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('disabled', (): void => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <IconButton disabled>
          <AcceptIcon />
        </IconButton>
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('size', (): void => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <IconButton size="small">
          <AcceptIcon />
        </IconButton>
        <IconButton size="medium">
          <AcceptIcon />
        </IconButton>
        <IconButton size="large">
          <AcceptIcon />
        </IconButton>
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('onClick', (): void => {
    const onClick = jest.fn()
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <IconButton onClick={onClick}>
          <AcceptIcon />
        </IconButton>
      </ThemeProvider>
    )
    const tree = component.toJSON()

    const iconButton = findAllByType(tree, 'button')
    iconButton[0].props.onClick()
    expect(onClick).toHaveBeenCalled()
  })

  test('as', (): void => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <IconButton as="a" href="#">
          <AcceptIcon />
        </IconButton>
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
