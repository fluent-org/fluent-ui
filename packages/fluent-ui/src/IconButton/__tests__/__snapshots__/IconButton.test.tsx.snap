// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton as 1`] = `
.c1 {
  width: 1em;
  height: 1em;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: inherit;
  color: inherit;
  fill: currentColor;
}

.c0 {
  border: none;
  outline: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  color: inherit;
  border-radius: 50%;
  overflow: hidden;
  -webkit-transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  color: #0078d4;
  background-color: transparent;
  padding: 12px;
}

.c0:disabled {
  color: #595959;
  cursor: not-allowed;
  pointer-events: none;
}

.c0:hover,
.c0:focus {
  background-color: rgba(36,149,224,0.08);
}

.c0:active {
  background-color: rgba(36,149,224,0.36);
}

<a
  className="c0"
  href="#"
>
  <svg
    className="c1"
    viewBox="0 0 2048 2048"
  >
    <path
      d="M640 1755L19 1133l90-90 531 530L1939 275l90 90L640 1755z"
    />
  </svg>
</a>
`;

exports[`IconButton basic 1`] = `
.c1 {
  width: 1em;
  height: 1em;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: inherit;
  color: inherit;
  fill: currentColor;
}

.c0 {
  border: none;
  outline: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  color: inherit;
  border-radius: 50%;
  overflow: hidden;
  -webkit-transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  color: #0078d4;
  background-color: transparent;
  padding: 12px;
}

.c0:disabled {
  color: #595959;
  cursor: not-allowed;
  pointer-events: none;
}

.c0:hover,
.c0:focus {
  background-color: rgba(36,149,224,0.08);
}

.c0:active {
  background-color: rgba(36,149,224,0.36);
}

<button
  className="c0"
>
  <svg
    className="c1"
    viewBox="0 0 2048 2048"
  >
    <path
      d="M640 1755L19 1133l90-90 531 530L1939 275l90 90L640 1755z"
    />
  </svg>
</button>
`;

exports[`IconButton disabled 1`] = `
.c1 {
  width: 1em;
  height: 1em;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: inherit;
  color: inherit;
  fill: currentColor;
}

.c0 {
  border: none;
  outline: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  color: inherit;
  border-radius: 50%;
  overflow: hidden;
  -webkit-transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  color: #0078d4;
  background-color: transparent;
  padding: 12px;
}

.c0:disabled {
  color: #595959;
  cursor: not-allowed;
  pointer-events: none;
}

.c0:hover,
.c0:focus {
  background-color: rgba(36,149,224,0.08);
}

.c0:active {
  background-color: rgba(36,149,224,0.36);
}

<button
  className="c0"
  disabled={true}
>
  <svg
    className="c1"
    viewBox="0 0 2048 2048"
  >
    <path
      d="M640 1755L19 1133l90-90 531 530L1939 275l90 90L640 1755z"
    />
  </svg>
</button>
`;

exports[`IconButton size 1`] = `
Array [
  .c1 {
  width: 1em;
  height: 1em;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: inherit;
  color: inherit;
  fill: currentColor;
}

.c0 {
  border: none;
  outline: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  color: inherit;
  border-radius: 50%;
  overflow: hidden;
  -webkit-transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  color: #0078d4;
  background-color: transparent;
  padding: 8px;
  font-size: 0.8em;
}

.c0:disabled {
  color: #595959;
  cursor: not-allowed;
  pointer-events: none;
}

.c0:hover,
.c0:focus {
  background-color: rgba(36,149,224,0.08);
}

.c0:active {
  background-color: rgba(36,149,224,0.36);
}

<button
    className="c0"
    size="small"
  >
    <svg
      className="c1"
      viewBox="0 0 2048 2048"
    >
      <path
        d="M640 1755L19 1133l90-90 531 530L1939 275l90 90L640 1755z"
      />
    </svg>
  </button>,
  .c1 {
  width: 1em;
  height: 1em;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: inherit;
  color: inherit;
  fill: currentColor;
}

.c0 {
  border: none;
  outline: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  color: inherit;
  border-radius: 50%;
  overflow: hidden;
  -webkit-transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  color: #0078d4;
  background-color: transparent;
  padding: 12px;
}

.c0:disabled {
  color: #595959;
  cursor: not-allowed;
  pointer-events: none;
}

.c0:hover,
.c0:focus {
  background-color: rgba(36,149,224,0.08);
}

.c0:active {
  background-color: rgba(36,149,224,0.36);
}

<button
    className="c0"
    size="medium"
  >
    <svg
      className="c1"
      viewBox="0 0 2048 2048"
    >
      <path
        d="M640 1755L19 1133l90-90 531 530L1939 275l90 90L640 1755z"
      />
    </svg>
  </button>,
  .c1 {
  width: 1em;
  height: 1em;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: inherit;
  color: inherit;
  fill: currentColor;
}

.c0 {
  border: none;
  outline: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  color: inherit;
  border-radius: 50%;
  overflow: hidden;
  -webkit-transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  color: #0078d4;
  background-color: transparent;
  padding: 16px;
  font-size: 1.2em;
}

.c0:disabled {
  color: #595959;
  cursor: not-allowed;
  pointer-events: none;
}

.c0:hover,
.c0:focus {
  background-color: rgba(36,149,224,0.08);
}

.c0:active {
  background-color: rgba(36,149,224,0.36);
}

<button
    className="c0"
    size="large"
  >
    <svg
      className="c1"
      viewBox="0 0 2048 2048"
    >
      <path
        d="M640 1755L19 1133l90-90 531 530L1939 275l90 90L640 1755z"
      />
    </svg>
  </button>,
]
`;

exports[`IconButton variant 1`] = `
Array [
  .c1 {
  width: 1em;
  height: 1em;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: inherit;
  color: inherit;
  fill: currentColor;
}

.c0 {
  border: none;
  outline: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  color: inherit;
  border-radius: 50%;
  overflow: hidden;
  -webkit-transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  color: #0078d4;
  background-color: transparent;
  padding: 12px;
}

.c0:disabled {
  color: #595959;
  cursor: not-allowed;
  pointer-events: none;
}

.c0:hover,
.c0:focus {
  background-color: rgba(36,149,224,0.08);
}

.c0:active {
  background-color: rgba(36,149,224,0.36);
}

<button
    className="c0"
  >
    <svg
      className="c1"
      viewBox="0 0 2048 2048"
    >
      <path
        d="M640 1755L19 1133l90-90 531 530L1939 275l90 90L640 1755z"
      />
    </svg>
  </button>,
  .c1 {
  width: 1em;
  height: 1em;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: inherit;
  color: inherit;
  fill: currentColor;
}

.c0 {
  border: none;
  outline: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  color: inherit;
  border-radius: 50%;
  overflow: hidden;
  -webkit-transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  transition: background-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,border-color 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,box-shadow 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms,padding 250ms cubic-bezier(0.76,0.24,0.23,0.94) 0ms;
  color: #ffffff;
  background-color: #0078d4;
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.22),0px 3px 8px 0px rgba(0,0,0,0.16),0px 1px 14px 0px rgba(0,0,0,0.1);
  padding: 12px;
}

.c0:disabled {
  color: #595959;
  cursor: not-allowed;
  pointer-events: none;
}

.c0:hover,
.c0:focus {
  background-color: #005cad;
}

.c0:active {
  color: #ffffff;
  background-color: #002d61;
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.22),0px 7px 11px 1px rgba(0,0,0,0.16),0px 3px 17px 2px rgba(0,0,0,0.1);
}

<button
    className="c0"
  >
    <svg
      className="c1"
      viewBox="0 0 2048 2048"
    >
      <path
        d="M640 1755L19 1133l90-90 531 530L1939 275l90 90L640 1755z"
      />
    </svg>
  </button>,
]
`;
